feat: Implement resuable ChatBubble component

choiseona opened the pull request:
## 📂 작업 내용

closes #13 

- [x] Default 컴포넌트 구현
- [x] Secondary 컴포넌트 구현
- [x] props로 variant, content, nickname 받기

## 💡 자세한 설명

채팅에 쓸 ChatBubble 컴포넌트를 구현했습니다. 

## 📗 참고 자료 & 구현 결과 (선택)

![image](https://github.com/user-attachments/assets/c1eed14c-b94b-427c-bbaa-48ddc42a96ce)

- 위의 이미지에서 닉네임이 흰색으로 보이지 않아 backgroundColor를 임의로 추가했습니다. 코드에서는 backgroundColor를 삭제한 상태입니다. 

## ✅ 셀프 체크리스트

- [x] PR 제목을 형식에 맞게 작성했나요?
- [x] 브랜치 전략에 맞는 브랜치에 PR을 올리고 있나요? (`main`이 아닙니다.)
- [x] 이슈는 close 했나요?
- [x] Reviewers, Labels를 등록했나요?
- [x] 작업 도중 문서 수정이 필요한 경우 잘 수정했나요?
- [x] 불필요한 코드는 제거했나요?

choiseona committed:
feat: Add resuable ChatBubble component

* Implement reusable ChatBubble component with class-variance-authority
* Add default and secondary variants
* Support conditional rendering based on nickname presence
* Add flexible layout with proper message alignment

choiseona committed:
refactor: Enhance ChatBubble component readability

- Simplify conditional rendering with isLeftAligned flag
- Consolidate duplicate layout structure
- Improve props destructuring for better readability

choiseona committed:
feat: Enhance ChatBubble component accessibility

- Add tabIndex for keyboard focus management
- Implement ARIA labels for screen reader support
- Set aria-hidden for nickname to prevent duplicate reading
- Add descriptive aria-labels to identify message sender and content

choiseona committed:
refactor: extract ariaLabel variable and simplify imports

- Extracted ariaLabel creation logic to a separate variable for better readability
- Replaced wildcard import with specific imports (orwardRef, HTMLAttributes) from 'react'

choiseona committed:
design: Remove background color

- Remove Chat backgroundColor

rhino-ty commented on client/src/components/ui/ChatBubble.tsx:
```diff
@@ -0,0 +1,55 @@
+        secondary: 'bg-chartreuseyellow-200',
+      },
+      size: {
+        default: 'text-base min-h-8',
```
`Input`에서도 말 하긴 했지만 `size`가 1개 밖에 없다면 그냥 인라인으로 넣는 것도 방법이 됩니당!

choiseona replied:
아래와 같이 공통 스타일로 통합해주었고,  

```js
const chatBubbleVariants = cva(
  'px-2.5 inline-flex items-center justify-center rounded-lg border-2 border-violet-950 text-violet-950 text-base min-h-8',
  {
    variants: {
      variant: {
        default: 'bg-halfbaked-200',
        secondary: 'bg-chartreuseyellow-200',
      },
    },
    defaultVariants: {
      variant: 'default',
    },
  },
);
```
ref도 제거하였습니다! 
```js
export interface ChatBubbleProps extends HTMLAttributes<HTMLDivElement>, VariantProps<typeof chatBubbleVariants> {
  content: string;
  nickname?: string;
}

const ChatBubble = ({ className, variant, content, nickname, ...props }: ChatBubbleProps) => {
  const isOtherUser = Boolean(nickname);
  const ariaLabel = isOtherUser ? `${nickname}님의 메시지: ${content}` : `내 메시지: ${content}`;

  return (
    <div
      aria-label={ariaLabel}
      tabIndex={0}
      className={cn('flex', isOtherUser ? 'flex-col items-start gap-0.5' : 'justify-end')}
    >
      {isOtherUser && (
        <span className="text-xs text-eastbay-50" aria-hidden="true">
          {nickname}
        </span>
      )}
      <p className={cn(chatBubbleVariants({ variant, className }))} {...props}>
        {content}
      </p>
    </div>
  );
};
```

choiseona committed:
refactor: remove ref and forwardRef, simplify styling

- Converted ChatBubble to a regular functional component by removing unnecessary ref and forwardRef
- Applied size styles directly as a single design system is in use
